base/linalg/matmul.jl

```diff
@@ -81,6 +81,13 @@ function (*){T,S}(A::AbstractMatrix{T}, x::AbstractVector{S})
     A_mul_B!(similar(x,TS,size(A,1)),A,x)
 end
 
+# these will throw a DimensionMismatch unless B has 1 row (or 1 col for transposed case):
+A_mul_B(a::AbstractVector, B::AbstractMatrix) = A_mul_B(reshape(a,length(a),1),B)
+A_mul_Bt(a::AbstractVector, B::AbstractMatrix) = A_mul_Bt(reshape(a,length(a),1),B)
+A_mul_Bt(A::AbstractMatrix, b::AbstractVector) = A_mul_Bt(A,reshape(b,length(b),1))
+A_mul_Bc(a::AbstractVector, B::AbstractMatrix) = A_mul_Bc(reshape(a,length(a),1),B)
```

Comment:
Since `RowVector` is an `AbstractMatrix`, it should always work with the generic fallback.